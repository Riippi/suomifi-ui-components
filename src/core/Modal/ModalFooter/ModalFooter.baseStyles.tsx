import { css } from 'styled-components';
import { defaultThemeTokens as theme } from '../../theme';

export const baseStyles = css`
  &.fi-modal_footer {
    flex: 0 0 auto;
    position: relative;
    & .fi-modal_footer_content {
      padding-top: 0;
      padding-right: ${theme.spacing.xl};
      padding-bottom: ${theme.spacing.m};
      padding-left: ${theme.spacing.s};
      & > * {
        margin-top: ${theme.spacing.m};
        margin-left: ${theme.spacing.s};
      }
    }

    & .fi-modal_footer_content-gradient-overlay {
      position: absolute;
      height: 51px;
      width: 100%;
      top: -51px;
      border-bottom: 1px solid ${theme.colors.depthLight1};
      & .fi-modal_footer_content-gradient {
        height: 50px;
        margin: 0 ${theme.spacing.m} 0 ${theme.spacing.m};
        background: linear-gradient(
          0deg,
          ${theme.colors.whiteBase},
          rgba(255, 255, 255, 0)
        );
      }
    }

    &--small-screen {
      & .fi-modal_footer_content {
        padding-top: 0;
        padding-right: ${theme.spacing.m};
        padding-bottom: ${theme.spacing.s};
        padding-left: ${theme.spacing.m};
        & > * {
          display: block;
          width: 100%;
          margin-top: ${theme.spacing.s};
          margin-left: 0;
        }
      }
    }
  }
`;
