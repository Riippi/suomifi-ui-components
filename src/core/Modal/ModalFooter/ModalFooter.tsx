import React, { Component } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import { HtmlDiv, HtmlDivProps } from '../../../reset';
import { ModalConsumer, ModalVariant, baseClassName } from '../Modal/Modal';
import { baseStyles } from './ModalFooter.baseStyles';

export interface ModalFooterProps extends HtmlDivProps {}

interface InternalModalFooterProps extends ModalFooterProps {
  modalVariant: ModalVariant;
}

export const footerClassName = `${baseClassName}_footer`;
const footerClassNames = {
  smallScreen: `${footerClassName}--small-screen`,
  button: `${footerClassName}_button`,
};

class BaseModalFooter extends Component<InternalModalFooterProps> {
  render() {
    const {
      children,
      className,
      modalVariant = 'default',
      ...passProps
    } = this.props;

    return (
      <HtmlDiv
        className={classnames(footerClassName, className, {
          [footerClassNames.smallScreen]: modalVariant === 'smallScreen',
        })}
        {...passProps}
      >
        {children}
      </HtmlDiv>
    );
  }
}

const StyledModalFooter = styled(BaseModalFooter)`
  ${baseStyles}
`;

/**
 * <i class="semantics" />
 * Use for showing modal content.
 * Props other than specified explicitly are passed on to outermost content div.
 */
export class ModalFooter extends Component<ModalFooterProps> {
  render() {
    return (
      <ModalConsumer>
        {({ variant }) => (
          <StyledModalFooter modalVariant={variant} {...this.props} />
        )}
      </ModalConsumer>
    );
  }
}
