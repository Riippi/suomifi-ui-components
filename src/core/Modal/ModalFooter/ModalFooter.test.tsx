import React from 'react';
import { render } from '@testing-library/react';
import { Modal } from '../';
import { ModalFooter, ModalFooterProps } from './ModalFooter';
import { Button } from '../../../';

describe('Basic ModalFooter', () => {
  const BasicModal = (props?: Partial<ModalFooterProps>) => (
    <Modal visible={true} usePortal={false}>
      <ModalFooter {...props} data-testid="modal-footer-id">
        <Button>OK</Button>
        <Button variant="secondary">Cancel</Button>
      </ModalFooter>
    </Modal>
  );

  it('should have given className', () => {
    const { getByTestId } = render(
      BasicModal({ className: 'test-class-name' }),
    );
    expect(getByTestId('modal-footer-id')).toHaveClass('test-class-name');
  });

  it('should match snapshot', () => {
    const { container } = render(BasicModal());
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('ModalFooter variant', () => {
  const SmallScreenModal = (
    <Modal visible={true} usePortal={false} variant="smallScreen">
      <ModalFooter data-testid="modal-footer-id">
        <Button>OK</Button>
        <Button variant="secondary">Cancel</Button>
      </ModalFooter>
    </Modal>
  );

  it('smallScreen should have correct classname', () => {
    const { getByTestId } = render(SmallScreenModal);
    expect(getByTestId('modal-footer-id').parentElement).toHaveClass(
      'fi-modal_footer--small-screen',
    );
  });
});
