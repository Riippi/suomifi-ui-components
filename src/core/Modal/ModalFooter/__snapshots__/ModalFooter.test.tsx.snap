// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic ModalFooter should match snapshot 1`] = `
.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c4:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c4::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c4::-webkit-inner-spin-button {
  height: auto;
}

.c4::-webkit-outer-spin-button {
  height: auto;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c1.fi-modal {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
}

.c1.fi-modal--no-portal {
  z-index: 100;
}

.c1.fi-modal .fi-modal_overlay {
  background-color: rgba(41,41,41,0.5);
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-modal .fi-modal_content-container {
  border-radius: 4px;
  background-color: hsl(0,0%,100%);
  border-top: 4px solid hsl(212,63%,45%);
  overflow: hidden;
  max-height: calc(100vh - 50px);
  min-height: 230px;
  width: 800px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1.fi-modal--no-scroll .fi-modal_content-container {
  width: 540px;
  min-height: 190px;
}

.c1.fi-modal--small-screen .fi-modal_content-container {
  border-radius: 0;
  width: 100%;
  height: 100%;
  max-height: 100%;
}

.c3.fi-modal_footer {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
}

.c3.fi-modal_footer .fi-modal_footer_content {
  padding-top: 0;
  padding-right: 15px;
  padding-bottom: 20px;
  padding-left: 30px;
}

.c3.fi-modal_footer .fi-modal_footer_content > * {
  margin-top: 20px;
  margin-right: 15px;
}

.c3.fi-modal_footer .fi-modal_footer_content-gradient-overlay {
  position: absolute;
  height: 51px;
  width: 100%;
  top: -51px;
  border-bottom: 1px solid hsl(202,7%,80%);
}

.c3.fi-modal_footer .fi-modal_footer_content-gradient-overlay .fi-modal_footer_content-gradient {
  height: 50px;
  margin: 0 20px 0 20px;
  background: linear-gradient( 0deg, hsl(0,0%,100%), rgba(255,255,255,0) );
}

.c3.fi-modal_footer--small-screen .fi-modal_footer_content {
  padding-right: 5px;
  padding-bottom: 15px;
  padding-left: 20px;
}

.c3.fi-modal_footer--small-screen .fi-modal_footer_content > * {
  display: block;
  width: calc(100% - 15px);
  margin-top: 15px;
}

.c5 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  padding: 10px 20px;
  min-height: 40px;
  color: hsl(0,0%,100%);
  background: linear-gradient(0deg,hsl(212,63%,37%) 0%,hsl(212,63%,45%) 100%);
  border-radius: 2px;
  text-align: center;
  text-shadow: 0 1px 1px rgba(0,53,122,0.5);
  cursor: pointer;
}

.c5:focus {
  outline: none;
  position: relative;
}

.c5:focus::after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c5:hover {
  background: linear-gradient(0deg,hsl(212,63%,45%) 0%,hsl(212,63%,49%) 100%);
}

.c5:active {
  background: hsl(212,63%,37%);
}

.c5.fi-button--disabled,
.c5[disabled],
.c5:disabled {
  text-shadow: 0 1px 1px rgba(41,41,41,0.5);
  background: linear-gradient(0deg,hsl(202,7%,67%) 0%,hsl(202,7%,80%) 100%);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: not-allowed;
}

.c5.fi-button--disabled::after {
  border: none;
  box-shadow: none;
}

.c5.fi-button--fullwidth {
  display: block;
  width: 100%;
}

.c5.fi-button--inverted {
  background: none;
  background-color: hsl(212,63%,45%);
  border: 1px solid hsl(0,0%,100%);
  text-shadow: none;
}

.c5.fi-button--inverted:hover {
  background: linear-gradient(-180deg,rgba(255,255,255,0.1) 0%,rgba(255,255,255,0) 100%);
}

.c5.fi-button--inverted:active {
  background: none;
  background-color: hsl(212,63%,45%);
}

.c5.fi-button--inverted.fi-button--disabled,
.c5.fi-button--inverted[disabled],
.c5.fi-button--inverted:disabled {
  opacity: 0.41;
  background: none;
  background-color: none;
}

.c5.fi-button--secondary {
  color: hsl(212,63%,45%);
  background: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(212,63%,45%);
  text-shadow: none;
}

.c5.fi-button--secondary:hover {
  background: linear-gradient(-180deg,hsl(202,7%,93%) 0%,rgba(255,255,255,0) 100%);
}

.c5.fi-button--secondary:active {
  background: none;
  background-color: hsl(202,7%,93%);
}

.c5.fi-button--secondary.fi-button--disabled,
.c5.fi-button--secondary[disabled],
.c5.fi-button--secondary:disabled {
  color: hsl(202,7%,67%);
  text-shadow: none;
  background: none;
  background-color: hsl(212,63%,98%);
  border-color: hsl(202,7%,67%);
}

.c5.fi-button--secondary-noborder {
  color: hsl(212,63%,45%);
  background: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(212,63%,45%);
  text-shadow: none;
  border: none;
  background-color: transparent;
}

.c5.fi-button--secondary-noborder:hover {
  background: linear-gradient(-180deg,hsl(202,7%,93%) 0%,rgba(255,255,255,0) 100%);
}

.c5.fi-button--secondary-noborder:active {
  background: none;
  background-color: hsl(202,7%,93%);
}

.c5.fi-button--secondary-noborder.fi-button--disabled,
.c5.fi-button--secondary-noborder[disabled],
.c5.fi-button--secondary-noborder:disabled {
  color: hsl(202,7%,67%);
  text-shadow: none;
  background: none;
  background-color: hsl(212,63%,98%);
  border-color: hsl(202,7%,67%);
}

.c5.fi-button--link {
  color: hsl(212,63%,45%);
  color: hsl(212,63%,45%);
  background: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(212,63%,45%);
  text-shadow: none;
  background: linear-gradient(0deg,hsl(215,100%,95%),hsl(215,100%,97%));
  border: none;
}

.c5.fi-button--link:hover {
  background: linear-gradient(-180deg,hsl(202,7%,93%) 0%,rgba(255,255,255,0) 100%);
}

.c5.fi-button--link:active {
  background: none;
  background-color: hsl(202,7%,93%);
}

.c5.fi-button--link.fi-button--disabled,
.c5.fi-button--link[disabled],
.c5.fi-button--link:disabled {
  color: hsl(202,7%,67%);
  text-shadow: none;
  background: none;
  background-color: hsl(212,63%,98%);
  border-color: hsl(202,7%,67%);
}

.c5.fi-button--link:hover {
  background: linear-gradient(0deg,hsl(215,100%,97%),hsl(212,63%,98%));
}

.c5.fi-button--link:active {
  background: linear-gradient(0deg,hsl(202,7%,93%),hsl(202,7%,97%));
}

.c5.fi-button--link.fi-button--disabled,
.c5.fi-button--link[disabled],
.c5.fi-button--link:disabled {
  color: hsl(202,7%,67%);
  background: none;
  background-color: hsl(202,7%,97%);
}

.c5 > .fi-button_icon {
  width: 16px;
  height: 16px;
  margin-right: 8px;
  vertical-align: middle;
  -webkit-transform: translateY(-0.1em);
  -ms-transform: translateY(-0.1em);
  transform: translateY(-0.1em);
}

.c5 > .fi-button_icon.fi-button_icon--right {
  margin-right: 0;
  margin-left: 8px;
}

.c5.fi-button--disabled > .fi-button_icon {
  cursor: not-allowed;
}

<div
  aria-describedby="3_title"
  aria-modal="true"
  class="c0 c1 fi-modal fi-modal--no-portal"
  role="dialog"
>
  <div
    class="c2 fi-modal_overlay"
  >
    <div
      class="c2 fi-modal_content-container"
    >
      <div
        class="c2 c3 fi-modal_footer"
      >
        <div
          class="c2 fi-modal_footer_content"
          data-testid="modal-footer-id"
        >
          <button
            aria-disabled="false"
            class="c4 fi-button c5"
            tabindex="0"
            type="button"
          >
            OK
          </button>
          <button
            aria-disabled="false"
            class="c4 fi-button c5 fi-button--secondary"
            tabindex="0"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          class="c2 fi-modal_footer_content-gradient-overlay"
        >
          <div
            class="c2 fi-modal_footer_content-gradient"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
